### PR TITLE
enhance: context menu budget page positioning

### PR DESCRIPTION
**Pull request by UnderKoen**
- Make the context menu's postioned under the mouse instead of static.
- Make the context menu non selectable.

#3706

### CODE REVIEW

File: upcoming-release-notes/3775.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+category: Enhancements
+authors: [UnderKoen]
+---
+
+Position context menu's on the to budget page to the cursor.
+And make popovers non selectable.
```
Comment (UnderKoen) [inline upcoming-release-notes/3775.md:7]:
```suggestion
Position context menu's on the to budget page to the cursor, and make popovers non selectable.
```

**Comment by youngcw** [comment]
Could you give some example repo steps?

**Comment by UnderKoen** [comment]
@youngcw 
Sorry I hadn't change the description. 

For reproduction of the selectable context menu. I don't know how @jfdoming achieved this https://github.com/actualbudget/actual/pull/3381#pullrequestreview-2397604856

**Comment by jfdoming** [comment]
> @youngcw Sorry I hadn't change the description.
> 
> For reproduction of the selectable context menu. I don't know how @jfdoming achieved this [#3381 (review)](https://github.com/actualbudget/actual/pull/3381#pullrequestreview-2397604856)

I believe I just hit Ctrl-A which selected the items

File: upcoming-release-notes/3775.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
+authors: [UnderKoen]
+---
+
+Position context menu's on the to budget page to the cursor, and make popovers non selectable.
```
Comment (matt-fidd) [inline upcoming-release-notes/3775.md:6]:
```suggestion
Position context menus on the to budget page to the cursor, and make popovers non selectable.
```

**Review by matt-fidd — CHANGES_REQUESTED** [review]
Just a spelling nit, then ready to merge

File: packages/desktop-client/src/hooks/useContextMenu.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    e.preventDefault();
+    setAsContextMenu(true);
+
+    const rect = e.currentTarget.getBoundingClientRect();
+    setPosition({
+      crossOffset: e.clientX - rect.left,
+      offset: e.clientY - rect.bottom,
+    });
+    setMenuOpen(true);
+  };
```
[the remark this answers is not in the document]
Comment (UnderKoen) [reply packages/desktop-client/src/hooks/useContextMenu.ts:23]: Popover does this for us

File: packages/desktop-client/src/components/transactions/TransactionsTable.jsx [hunk 4]
```diff
@@ -1081,23 +1079,14 @@ const Transaction = memo(function Transaction({
...
-      onContextMenu={e => {
-        if (!contextMenusEnabled) return;
-        if (transaction.id === 'temp') return;
-        e.preventDefault();
-        const rect = e.currentTarget.getBoundingClientRect();
-        setCrossOffset(e.clientX - rect.left);
-        setOffset(e.clientY - rect.bottom);
-        setMenuOpen(true);
-      }}
+      onContextMenu={handleContextMenu}
```
[the remark this answers is not in the document]
Comment (UnderKoen) [reply packages/desktop-client/src/components/transactions/TransactionsTable.jsx:1082]: It does

**Review by matt-fidd — APPROVED** [review]
LGTM!

**Review by matt-fidd — CHANGES_REQUESTED** [review]
Ah sorry, couple of lint errors to fix first

**Review by matt-fidd — APPROVED** [review]

**Pull request merged**